src/System.Runtime/tests/System/NullableTests.cs

```diff
@@ -113,6 +113,47 @@ public static void Compare_Equals(int? n1, int? n2, int expected)
             Assert.Equal(expected, Nullable.Compare(n1, n2));
         }
 
+        [Fact]
+        public static void MutatingMethods_MutationsAffectOriginal()
+        {
+            MutatingStruct? ms = new MutatingStruct() { Value = 1 };
+
+            for (int i = 1; i <= 2; i++)
+            {
+                Assert.Equal(i.ToString(), ms.Value.ToString());
+                Assert.Equal(i, ms.Value.Value);
+
+                Assert.Equal(i.ToString(), ms.ToString());
+                Assert.Equal(i + 1, ms.Value.Value);
+            }
+
+            for (int i = 3; i <= 4; i++)
+            {
+                Assert.Equal(i, ms.Value.GetHashCode());
```

Comment:
This test seems to rely on the fact that `GetHashCode()` for a value of type `int` just returns the value itself. I would say this is an implementation detail that this test should not depend on. I propose to change this line to `Assert.Equal(i.GetHashCode(), ms.Value.GetHashCode())`, or, alternatively, remove the call to `GetHashCode()` from `Value++.GetHashCode()` on line 154 in the implementation of `MutatingStruct`.

Reply:
> I would say this is an implementation detail

It's a detail we have tests for elsewhere.  There's no problem having it here.  If it ever changed (and I highly doubt it will), we can change this as well as a multitude of other tests that rely on it.